src/System/Random/Internal.hs

```diff
@@ -709,34 +713,47 @@ instance UniformRange Bool where
 -- | See /Floating point number caveats/ in "System.Random.Stateful".
 instance UniformRange Double where
   uniformRM (l, h) g = do
-    w64 <- uniformWord64 g
-    let x = word64ToDoubleInUnitInterval w64
+    x <- uniformDouble01M g
     return $ (h - l) * x + l
 
--- | Turns a given uniformly distributed 'Word64' value into a uniformly
--- distributed 'Double' value in the range [0, 1].
-word64ToDoubleInUnitInterval :: Word64 -> Double
-word64ToDoubleInUnitInterval w64 = d / m
+-- | Generate uniformly distributed 'Double' in the range [0, 1].
+uniformDouble01M :: StatefulGen g m => g -> m Double
+uniformDouble01M g = do
+  w64 <- uniformWord64 g
+  return $ fromIntegral w64 / m
   where
-    d = fromIntegral w64 :: Double
     m = fromIntegral (maxBound :: Word64) :: Double
-{-# INLINE word64ToDoubleInUnitInterval #-}
+
+-- | Generate uniformly distributed 'Double' in the range (0, 1]. That
+--   is result is guaranteed to be positive
+uniformDoublePos01M :: StatefulGen g m => g -> m Double
```

Comment:
When I see "pos", I think "position". How about `uniformDoublePositive01M`? Longer but unambiguous.

Additional thought: "positive 0" actually refers to `+0`, as opposed to "negative zero", `-0`, in IEEE 754. So people might think that something called `uniformDoublePos01M` or `uniformDoublePositive01M` includes positive zero. To avoid "positive" and "0" following each other in the name of the function, perhaps the scheme could be

- `uniformDouble01`
- `uniformDouble01Positive`
- `uniformFloat01`
- `uniformFloat01Positive`